### PR TITLE
Month Year Field Dialog Part One

### PR DESCRIPTION
**Pull request by juantobonq**
UI Test:
image

### CODE REVIEW

File: app/src/main/java/co/condorlabs/emerald/components/monthyearpicker/EmeraldMonthYearPicker.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    dismissText: String = stringResource(id = R.string.cancel),
+    onDismissButton: () -> Unit,
+    onConfirmButton: (Int, Int) -> Unit,
+) {
+
+    val months = stringArrayResource(id = R.array.months)
+    var state by remember { mutableStateOf(months[monthSelected]) }
+    var pickerValue by remember { mutableStateOf(yearSelected) }
+
+    if (isShowDialog) {
```
[the remark this answers is not in the document]
Comment (juantobonq) [reply app/src/main/java/co/condorlabs/emerald/components/monthyearpicker/EmeraldMonthYearPicker.kt]: Done

File: app/src/androidTest/java/co/condorlabs/emerald/emeraldmonthyeardialog/EmeraldMonthYearDialogPage.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.Calendar
+
+class EmeraldMonthYearDialogPage(private val composeRule: ComposeContentTestRule) : Page() {
+
+    fun launchView(
+        showDialog : Boolean = true,
+        monthSelected: Int = Calendar.getInstance().get(Calendar.MONTH),
+        yearSelected: Int = Calendar.getInstance().get(Calendar.YEAR),
+        onDismissButton: () -> Unit = {},
+        onConfirmButton: (Int, Int) -> Unit = { _, _ ->},
```
Comment (cljacob) [inline app/src/androidTest/java/co/condorlabs/emerald/emeraldmonthyeardialog/EmeraldMonthYearDialogPage.kt:19]: the comma at the end is necessary?

File: app/src/androidTest/java/co/condorlabs/emerald/emeraldmonthyeardialog/EmeraldMonthYearDialogPage.kt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    ) = apply {
+        composeRule.setContent {
+            EmeraldTheme {
+                EmeraldMonthYearPicker(
+                    label = "Month year",
+                    showDialog = showDialog,
+                    monthSelected = monthSelected,
+                    yearSelected = yearSelected,
+                    onDismissButton = onDismissButton,
+                    onConfirmButton = onConfirmButton,
```
Comment (cljacob) [inline app/src/androidTest/java/co/condorlabs/emerald/emeraldmonthyeardialog/EmeraldMonthYearDialogPage.kt:29]: the comma at the end is necessary?

File: app/src/main/java/co/condorlabs/emerald/components/monthyearpicker/EmeraldMonthYearPicker.kt [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    showDialog: Boolean = false,
+    minYear: Int = integerResource(id = R.integer.min_year),
+    maxYear: Int = integerResource(id = R.integer.max_year),
+    monthSelected: Int = Calendar.getInstance().get(Calendar.MONTH),
+    yearSelected: Int = Calendar.getInstance().get(Calendar.YEAR),
+    label: String = Empty,
+    confirmText: String = stringResource(id = R.string.ok),
+    dismissText: String = stringResource(id = R.string.cancel),
+    onDismissButton: () -> Unit,
+    onConfirmButton: (Int, Int) -> Unit,
```
Comment (cljacob) [inline app/src/main/java/co/condorlabs/emerald/components/monthyearpicker/EmeraldMonthYearPicker.kt:39]: the comma at the end is necessary?

Comment (juantobonq) [reply app/src/main/java/co/condorlabs/emerald/components/monthyearpicker/EmeraldMonthYearPicker.kt:39]: Done

File: app/src/androidTest/java/co/condorlabs/emerald/emeraldmonthyeardialog/EmeraldMonthYearDialogPage.kt (hunk 3 above)
Comment (juantobonq) [reply app/src/androidTest/java/co/condorlabs/emerald/emeraldmonthyeardialog/EmeraldMonthYearDialogPage.kt:29]: Done

File: app/src/androidTest/java/co/condorlabs/emerald/emeraldmonthyeardialog/EmeraldMonthYearDialogPage.kt (hunk 2 above)
Comment (juantobonq) [reply app/src/androidTest/java/co/condorlabs/emerald/emeraldmonthyeardialog/EmeraldMonthYearDialogPage.kt:19]: Done

**Review by cljacob — APPROVED** [review]

**Review by JhoanGamarra — APPROVED** [review]

**Pull request merged**